internal/daemon/api_files.go

```diff
@@ -538,22 +546,25 @@ func makeDir(dir makeDirsItem) error {
 	if err != nil {
 		return err
 	}
-	if dir.MakeParents {
-		err = os.MkdirAll(dir.Path, perm)
-	} else {
-		err = os.Mkdir(dir.Path, perm)
-	}
-	if err != nil {
-		return err
-	}
 	uid, gid, err := normalizeUidGid(dir.UserID, dir.GroupID, dir.User, dir.Group)
 	if err != nil {
 		return fmt.Errorf("cannot look up user and group: %w", err)
 	}
-	if uid != nil && gid != nil {
-		err = chown(dir.Path, *uid, *gid)
+	if dir.MakeParents {
+		err := mkdirAllUserGroup(dir.Path, perm, uid, gid)
+		if err != nil {
+			return err
+		}
+	} else {
+		err := os.Mkdir(dir.Path, perm)
```

Comment:
This should be implemented in the osutil package so that it can use the same mutex as the MkdirAllUserGroup and also employ the same Mkdir with temp name, Chown and Rename to final name logic.

Reply:
Done, thanks